src/olympia/reviewers/utils.py

```diff
@@ -828,21 +829,26 @@ def process_public(self):
         if self.set_addon_status:
             self.set_addon(status=amo.STATUS_APPROVED)
 
-        # Clear needs_human_review flags on past listed versions.
         if self.human_review:
+            # No need for a human review anymore.
             self.unset_past_needs_human_review()
-            # Clear the "needs_human_review" scanner flags too, if any, and
-            # only for the specified version.
             VersionReviewerFlags.objects.filter(
                 version=self.version
             ).update(needs_human_review_by_mad=False)
 
-        # Increment approvals counter if we have a request (it means it's a
-        # human doing the review) otherwise reset it as it's an automatic
-        # approval.
-        if self.human_review:
+            # An approval took place so we can reset this.
+            AddonReviewerFlags.objects.update_or_create(
```

Comment:
Why do we use `.update_or_create()` here? Instead of `.update()`?

Reply:
In this case both could work since we consider `False`, `None` and the absence of an `AddonReviewerFlags` instance as the same thing. In general though, I think `update_or_create` is slightly safer, since it will create the flags and set the explicit value you asked for no matter what. We haven't been very consistent with this, maybe we should...